src/engines/html5/media-source/adapters/native-adapter.js

```diff
@@ -1,57 +1,50 @@
 //@flow
-import LoggerFactory from '../../../../utils/logger'
 import EventManager from '../../../../event/event-manager'
-import PlayerEvents from '../../../../event/events'
+import {HTML5_EVENTS as Html5Events} from '../../../../event/events'
 import Track from '../../../../track/track'
 import VideoTrack from '../../../../track/video-track'
 import AudioTrack from '../../../../track/audio-track'
 import TextTrack from '../../../../track/text-track'
+import BaseMediaSourceAdapter from '../base-media-source-adapter'
+
 /**
  * An illustration of media source extension for progressive download
  * @classdesc
  * @implements {IMediaSourceAdapter}
  */
-export default class NativeAdapter implements IMediaSourceAdapter {
+export default class NativeAdapter extends BaseMediaSourceAdapter {
   /**
    * The name of the Adapter
    * @member {string} _name
    * @static
    * @private
    */
-  static _name = "NativeAdapter";
+  static _name: string = 'NativeAdapter';
   /**
    * Getter for the adapter name
    * @returns {string} - The adapter name
+   * @static
    */
   static get name(): string {
     return NativeAdapter._name;
   }
 
   /**
-   * The adapter logger
-   * @member {any} _logger
-   * @private
+   * @param {string} name - The adapter name.
+   * @returns {void}
    * @static
    */
-  static _logger = LoggerFactory.getLogger(NativeAdapter._name);
-  /**
-   * The adapter config
-   * @member {Object} _config
-   * @private
-   */
-  _config: Object;
-  /**
-   * The source object
-   * @member {Source} _sourceObj
-   * @private
-   */
-  _sourceObj: ?Source;
+  static set name(name: string): void {
+    // Do nothing. Just a workaround for flow issue with static getter in an inheritor. See: https://github.com/facebook/flow/issues/3008.
```

Comment:
done.
(no need getter and setter at all actually, just to define _name in the derived adapter)